addons/mail/models/mail_presence.py

```diff
@@ -94,15 +89,16 @@ def _update_presence(self, inactivity_period, identity_field, identity_value):
     def _invalidate_im_status(self):
         self.user_id.invalidate_recordset(["im_status"])
         self.user_id.partner_id.invalidate_recordset(["im_status"])
+        self.guest_id.invalidate_recordset(["im_status"])
 
     def _send_presence(self, im_status=None, bus_target=None):
         """Send notification related to bus presence update.
 
         :param im_status: 'online', 'away' or 'offline'
         """
         for presence in self:
-            identity_data = presence._get_identity_data()
-            target = presence._get_bus_target()
+            identity_data = {"guest_id": self.guest_id.id} if presence.guest_id else {"partner_id": presence.user_id.partner_id.id}
```

Comment:
it should be `presence.guest_id.id`/`presence.user_id.partner_id.id`. also, there  will always be a target now that we do not provide an override point so both the target and the identity_data variables are useless? And the condition can be removed.